syntax = "proto3";

option optimize_for = LITE_RUNTIME;

package KCPP.LCDStyle;

message LCDColour3 {
	float r = 1;
	float g = 2;
	float b = 3;
}

message LCDColour4 {
	float r = 1;
	float g = 2;
	float b = 3;
	float a = 4;
}

enum LCDUserNameType {
	UserName = 0;
	DisplayName = 1;
}

enum LCDElementAlignment {
	LEFT = 0;
	RIGHT = 1;
}

enum LCDElementAttachment {
	ATTACHED = 0;
	DETACHED = 1;
}

// backwards compat
message LCDColour3Double {
	double r = 1;
	double g = 2;
	double b = 3;
}

message LCDColour4Double {
	double r = 1;
	double g = 2;
	double b = 3;
	double a = 4;
}

message LCDStyleSave {
	LCDColour3 active_colour = 11;
	LCDColour3 prestige_active_colour = 12;
	LCDColour4 inactive_colour = 13;
	LCDColour4 bg_colour = 14;

	oneof user_name {
		string user_name_text = 4;
		LCDUserNameType user_name_policy = 5;
	}
	
	LCDElementAlignment lcd_grab_bar_alignment = 6;
	LCDElementAttachment lcd_grab_bar_attachment = 7;
	LCDElementAlignment lcd_prestige_counter_alignment = 8;
	LCDElementAttachment lcd_prestige_counter_attachment = 9;

	bool fast_quit = 16;

	uint32 lcd_save_version = 15;

	// backwards compat
	optional LCDColour3Double active_colour_double = 1;
	optional LCDColour3Double prestige_active_colour_double = 10;
	optional LCDColour4Double inactive_colour_double = 2;
	optional LCDColour4Double bg_colour_double = 3;
}